### PR TITLE
[feat][feat]: convert asset 

### PR DESCRIPTION
**Pull request by abc-alpaca**
## Description
convert asset action

### CODE REVIEW

File: contracts/8.10/protocol/DeltaNeutralVaultConfig.sol [hunk 1]
```diff
@@ -30,11 +30,19 @@ contract DeltaNeutralVaultConfig is IDeltaNeutralVaultConfig, OwnableUpgradeable
   );
   event LogSetWhitelistedCallers(address indexed _caller, address indexed _address, bool _ok);
   event LogSetWhitelistedRebalancers(address indexed _caller, address indexed _address, bool _ok);
+  event LogSetWhitelistedConvertAssetTokens(address indexed _caller, address indexed _address, bool _ok);
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/DeltaNeutralVaultConfig.sol:33]: should we remove this event?

**Review by sora-alpaca — CHANGES_REQUESTED** [review]

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 2]
```diff
@@ -621,6 +634,38 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
     SafeERC20Upgradeable.safeTransfer(IERC20Upgradeable(alpacaToken), _to, _amount);
   }
 
+  /// @notice convert Asset to asset
+  /// @dev convert asset by type
+  /// @param _value native value
+  /// @param _data abi_code data
+  function _convertAsset(uint256 _value, bytes memory _data) internal {
+    (uint256 _swapType, uint256 _amount, uint256 _amountOut, address _sourceToken, address _destinationToken) = abi
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:642]:
```suggestion
    (uint256 _swapType, uint256 _amountIn, uint256 _amountOut, address _sourceToken, address _destinationToken) = abi
```

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 3]
```diff
@@ -621,6 +634,38 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
...
+
+    address[] memory paths = config.getSwapRoutePathsAddr(_sourceToken, _destinationToken);
+
+    if (routerAddr == address(0) || paths.length == 0) {
+      revert InvalidConvertTokenSetting();
+    }
+
+    SafeERC20Upgradeable.safeApprove(IERC20Upgradeable(_sourceToken), routerAddr, type(uint256).max);
+    if (_swapType == CONVERT_EXACT_TOKEN_TO_NATIVE) {
+      ISwapRouter(routerAddr).swapExactTokensForETH(_amount, _amountOut, paths, address(this), block.timestamp);
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:655]:
```suggestion
      ISwapRouter(routerAddr).swapExactTokensForETH(_amountIn, _amountOut, paths, address(this), block.timestamp);
```

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 4]
```diff
@@ -621,6 +634,38 @@ contract DeltaNeutralVault is ERC20Upgradeable, ReentrancyGuardUpgradeable, Owna
...
+    }
+    if (_swapType == CONVERT_EXACT_NATIVE_TO_TOKEN) {
+      ISwapRouter(routerAddr).swapExactETHForTokens{ value: _value }(_amountOut, paths, address(this), block.timestamp);
+    }
+    if (_swapType == CONVERT_EXACT_TOKEN_TO_TOKEN) {
+      ISwapRouter(routerAddr).swapExactTokensForTokens(_amount, _amountOut, paths, address(this), block.timestamp);
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:661]:
```suggestion
      ISwapRouter(routerAddr).swapExactTokensForTokens(_amountIn, _amountOut, paths, address(this), block.timestamp);
```

**Review by alpanaca — APPROVED** [review]

**Pull request merged**